src/core/Akka/Actor/Internal/ActorSystemImpl.cs

```diff
@@ -225,6 +225,17 @@ public void Start()
                 {
                     _log.Info(Settings.ToString());
                 }
+
+                //HACK: ensure provider started
+                {
```

Comment:
Why is this necessary? Just to stop NREs from being raised by the remoting system? I don't like this - the idea behind `_provider.DefaultAddress` possibly returning `null` or throwing is that those are handled exceptions that occur as a result of the underlying transport not yet being bound, which is going to happen. Blocking and forcing the transport before we start creating actors is going to create problems. I'd like to see this removed unless there's a really good reason to keep it.

Reply:
The akka-system startup is a complete mess, we removed 2-3 locks with our last commits 
and akka.remote got completely racy, so many components rely on `Provider.DefaultAddress` 
and just NRE in the unit tests, 

this hack comes more to the behavior of version 1.4.21

Reply:
We should probably rewrite, maybe for Akka.NET 2.0 or 1.5, `ActorSystem.Create` to return a `Task` and just `await` all of the startup functions that occur inside the `ActorRefProvider` - so once an `ActorSystem` is started it's either really started or the startup failed and an exception was thrown right away.

Reply:
But as far as this PR goes - I guess we can live with this